src/modules/dashboard/sagas/utils/safeHelpers.ts

```diff
@@ -280,6 +281,92 @@ export const getTransferFundsData = async (
   ]);
 };
 
+const getParsedJSONOrArrayFromString = (
+  parameter: string,
+): (string | number)[] | null => {
+  try {
+    const arrayResult = JSON.parse(parameter);
+    return arrayResult.map((value) => {
```

Comment:
I believe JSON.parse also introduces rounding errors for large numbers. E.g. 2 ** 255 - 1 (the largest safe int) will be turned into 
`5.78960446186581e+76`, which does not `===` 2 ** 255 - 1. 

Whereas `getArrayFromString` will give you an array of the numbers as strings, so you don't need to convert to a big number and then back to a string.